### DISCUSSION TITLE
Inheritable Data Type Configurations

### DISCUSSION THREAD

**Discussion by NikRimington** [umbraco/Umbraco-CMS#12675] [Features and ideas]
Hi Everyone,

I was thinking about this recently and wondered if it would even be possible and what the general thought around it would be.

With controls like The Grid, Nested Content, Block List editor etc, would be possible to introduce the ability to have a "Base config". A bit like composition from a doc type perspective.

For example:

Block List A

 * This has 4 block types configured

Block List B  - Inherits from Block List A
* Adds 2 other block types and a different layout option.

With page builders where people have "mostly" the same blocks but on some types different/specific ones are needed, this would save the need to completely duplicate a data type.

Hope that makes sense, would love to know your thoughts :-)

**Comment by skttl** [comment]
Yes, this is how it should be!

**Comment by prjseal** [comment]
💯

**Comment by markadrake** [comment]
If we can have "content templates" maybe we can also have "document type templates" and "data type templates" and so forth?

**Comment by DanDiplo** [comment]
Yes, this would be great. I often have situations where I have various Block List datatypes that are essentially the same but with a few variations. Having a base datatype for the common blocks would be really helpful. The reason I wrote the "copy Datatype" thing for God Mode was to ease this pain, but this would be a better solution.
